corehq/apps/app_manager/models.py

```diff
@@ -3507,6 +3498,31 @@ def get_filter_value(self, user, ui_filter):
         return None
 
 
+class AncestorLocationTypeFilter(ReportAppFilter):
+    ancestor_location_type_name = StringProperty()
+
+    def get_filter_value(self, user, ui_filter):
+        """
+        A list of locations who have the same ancestor as user param
+
+        :raises LocationType.DoesNotExist: if ancestor_location_type_name does not match a location type
+        """
+        from corehq.apps.locations.models import LocationType
+        from corehq.apps.reports_core.filters import Choice
+
+        ancestor_type = LocationType.objects.get(name=self.ancestor_location_type_name)
```

Comment:
might need to guard against invalid user input here. I think this will raise a `NotFound` if they specify a bad type